File: python/0168.py
# Link: https://leetcode.com/problems/excel-sheet-column-title/description/
# Runtime: 34 ms
# Memory: 16.34 MB


class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        number = columnNumber
        column_title = ""
        while number != 0:
            number -= 1
            column_title = chr(ord("A") + number % 26) + column_title
            number //= 26

        return column_title

File: c++/0168.cpp
// Link: https://leetcode.com/problems/excel-sheet-column-title/description/
// Runtime: 2 ms
// Memory: 5.91 MB

class Solution {
public:
  std::string convertToTitle(int columnNumber) {
    int number = columnNumber;
    std::string columnTitle = "";
    while (number > 0) {
      number--;
      columnTitle = static_cast<char>('A' + number % 26) + columnTitle;
      number /= 26;
    }

    return columnTitle;
  }
};
